magefile.go

```diff
@@ -0,0 +1,77 @@
+//go:build mage
+// +build mage
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+const BIN_DIR = "bin"
+
+var (
+	Default  = All
+	goexe    = "go"
+	gofmtexe = "gofmt"
+)
+
+func init() {
+	if exe := os.Getenv("GOEXE"); exe != "" {
+		goexe = exe
+		gofmtexe = "gofmt.exe"
+	}
+}
+
+func All() {
+	Generate()
+	Format()
+	Build()
+	Test()
+	Install()
+}
+
+func Generate() error {
+	fmt.Println("⚙️  Running generators...")
+	cmd := exec.Command(goexe, "run", filepath.Join(".", "function-metadata", "extract.go"))
+	cmd.Stdout = os.Stdout
+	return cmd.Run()
+}
+
+func Format() error {
+	fmt.Println("⚙️  Formatting files...")
+	cmd := exec.Command(gofmtexe, "-s", "-w", ".")
+	cmd.Stdout = os.Stdout
+	return cmd.Run()
+}
+
+func Build() error {
+	fmt.Println("Building...")
+	mdir := exec.Command("mkdir", filepath.Join(".", BIN_DIR))
```

Comment:
Just flagging the `-p` is missing, but suppose will be replaced anyway when we use Go APIs.